contrib/endpoints/src/api_manager/context/service_context.cc

```diff
@@ -76,6 +76,18 @@ ServiceContext::ServiceContext(std::unique_ptr<ApiManagerEnvInterface> env,
 
   service_account_token_.SetAudience(
       auth::ServiceAccountToken::JWT_TOKEN_FOR_FIREBASE, kFirebaseAudience);
+
+  if (config_->server_config() &&
+      !config_->server_config()
+           ->api_check_security_rules_config()
+           .authorization_service_audience()
+           .empty()) {
+    service_account_token_.SetAudience(
+        auth::ServiceAccountToken::JWT_TOKEN_FOR_AUTHORIZATION_SERVICE,
```

Comment:
While that was what I had initially suggested we do about the audiences, I started having second thoughts once I started writing the firebase rules. For example, the fact that we are using something called "audiences" to generated an auth token is ESP specific and we are only generating these for Google based services. I am not sure how this will translate to custom auth cases.  I think we should think more about this and ask experts on the best way to do this. But this change should not affect this PR. We can address this as a part of another PR .